src/Network/GRPC/MQTT/RemoteClient.hs

```diff
@@ -204,34 +203,37 @@ pattern AuxMessageTerminate = AuxControlMessage (Enumerated (Right AuxControlTer
 -- | Handles AuxControl signals from the "/control" topic
 handleControlMessage :: Logger -> SessionHandle -> LByteString -> IO ()
 handleControlMessage logger handle msg = do
-  case fromLazyByteString msg of
-    Right AuxMessageTerminate -> do
-      Logger.logInfo logger "Received terminate message"
-      Async.cancel (hdlThread handle)
-    Right AuxMessageAlive -> do
-      Logger.logDebug logger "Received heartbeat message"
-      atomically do
-        _ <- tryPutTMVar (hdlHeartbeat handle) ()
-        pure ()
-    Right ctrl ->
-      Logger.logWarn logger $ "Received unknown control message: " <> show ctrl
-    Left err ->
-      Logger.logErr logger $ "Failed to parse control message: " <> show err
+  let result :: Either Decode.ParseError AuxControlMessage
+      result = Proto3.fromByteString (toStrict msg)
+   in case result of
+        Right AuxMessageTerminate -> do
+          Logger.logInfo logger "Received terminate message"
+          Async.cancel (hdlThread handle)
+        Right AuxMessageAlive -> do
+          Logger.logDebug logger "Received heartbeat message"
+          atomically do
+            _ <- tryPutTMVar (hdlHeartbeat handle) ()
+            pure ()
+        Right ctrl -> do
+          -- TODO: cancel hdlThread and abort
```

Comment:
Is it the case that we need to cancel the `hdlThread` on an unknown control message? I don't think we want to cancel the thread because we may receive other control messages that we do know how to handle.

Same goes for the todo comment on the `Left err` case too.

Reply:
Good point! I didn't think about that.